Avoid pinning dependencies, use --precise in ci

danielabrozzoni opened the pull request:
Fixes #1035

### Checklists

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

danielabrozzoni committed:
ci: Avoid pinning dependencies, use --precise

Fixes #1035

adoerr commented on crates/bdk/Cargo.toml:
```diff
@@ -13,7 +13,7 @@ edition = "2021"
 
 [dependencies]
-log = "=0.4.18"
+log = "^0.4"
```
Since a caret requirement is just another syntax for the default strategy, I would simply use `"0.4"` for consistency reasons

notmandatory commented:
LGTM. One thing to add is a short MSRV section in `bdk/README.md` similar to what the `rust-bitcoin` project did (https://github.com/rust-bitcoin/rust-bitcoin/blob/master/README.md#minimum-supported-rust-version-msrv). For example:

## Minimum Supported Rust Version (MSRV)

This library should compile with any combination of features with **Rust 1.57.0**.

To build with the MSRV you will need to pin dependencies as follows:

```
# log 0.4.19 has MSRV 1.60.0+
cargo update -p log --precise "0.4.18"
# tempfile 3.7.0 has MSRV 1.63.0
cargo update -p tempfile --precise "3.6.0"
```

notmandatory commented on .github/workflows/cont_integration.yml:
```diff
@@ -27,6 +27,9 @@ jobs:
             profile: minimal
       - name: Rust Cache
         uses: Swatinem/rust-cache@v2.2.1
+      - name: Pin dependencies
```
```suggestion
      - name: Pin dependencies for MSRV
```

danielabrozzoni committed:
ci: No need to add the llvm repository for wasm

Adding it fails with "The repository'http://apt.llvm.org/focal
llvm-toolchain-focal Release' does not have a Release file.", and for
some reason it's not needed anymore.

danielabrozzoni commented:
Thanks! Fixed all your comments, and added a fix for the wasm build.

adoerr reviewed (approved):
LGTM

evanlinjin reviewed (approved):
ACK ffb7c795e15d2d879f2d857e7607729ca4a53155
